crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -930,6 +931,50 @@ impl EntityCommands<'_> {
         self.add(insert(bundle))
     }
 
+    /// Adds a dynamic component to an entity.
+    ///
+    /// See [`EntityWorldMut::insert_by_id`] for more information.
+    ///
+    /// # Panics
+    ///
+    /// The command will panic when applied if the associated entity does not exist.
+    ///
+    /// To avoid a panic in this case, use the command [`Self::try_insert`] instead.
+    ///
+    /// # Safety
+    ///
+    /// - [`ComponentId`] must be from the same world as `self`.
+    /// - `T` must have the same layout as the one passed during `componennt_id` creation.
+    pub unsafe fn insert_by_id<T: Send + 'static>(
+        &mut self,
+        component_id: ComponentId,
+        value: T,
+    ) -> &mut Self {
+        // SAFETY: same invariants as parent call
+        self.add(unsafe {insert_by_id(component_id, value, move |entity| {
+            panic!("error[B0003]: Could not insert a component {component_id:?} (with type {}) for entity {:?} because it doesn't exist in this World. See: https://bevyengine.org/learn/errors/#b0003", std::any::type_name::<T>(), entity);
+        })});
+        self
+    }
+
+    /// Attempts to add a dynamic component to an entity.
```

Comment:
suggestion: Should probably mention what happens if the entity doesn't exist (i.e., the attempt is silently ignored), since it has a `try_`-prefix but doesn't return a `Result<>` or similar.
The behavior is only explicitly mentioned in the `insert_by_id()` docs.

thought(non-blocking): Maybe this should not completely ignore the failure, but log it using `warn!/debug!/trace!`?
But don't know how such cases are handled in bevy usually, best to stick to common practices here.